protocol/test/forge/v2/treasuryReservesVault/TrvBase.t.sol

```diff
@@ -0,0 +1,421 @@
+        _setUp();
+    }
+
+    function test_initalization() public {
+        assertEq(trv.apiVersion(), "1.0.0");
+        address[] memory strategiesList = trv.strategiesList();
+        assertEq(strategiesList.length, 0);
+        address[] memory tokensList = trv.borrowTokensList();
+        assertEq(tokensList.length, 0);
+        assertEq(trv.globalBorrowPaused(), false);
+        assertEq(trv.globalRepaysPaused(), false);
+        assertEq(address(trv.tpiOracle()), address(tpiOracle));
+        assertEq(trv.treasuryPriceIndex(), 0.97e18);
+    }
+
+    function test_setBorrowToken() public {
+        vm.startPrank(executor);
+
+        vm.expectRevert(abi.encodeWithSelector(CommonEventsAndErrors.InvalidAddress.selector));
+        trv.setBorrowToken(IERC20(address(0)), address(0), 0, 0, address(0));
+
+        vm.expectRevert(abi.encodeWithSelector(CommonEventsAndErrors.InvalidAddress.selector));
+        trv.setBorrowToken(dai, address(0), 0, 0, address(0));
+
+        vm.expectRevert(abi.encodeWithSelector(CommonEventsAndErrors.InvalidParam.selector));
+        trv.setBorrowToken(dai, address(0), 100, 99, address(dUSD));
+
+        // Base strategy left unset
+        vm.expectEmit(address(trv));
+        emit BorrowTokenSet(address(dai), address(0), 100, 101, address(dUSD));
+        trv.setBorrowToken(dai, address(0), 100, 101, address(dUSD));
+
+        address[] memory tokensList = trv.borrowTokensList();
+        assertEq(tokensList.length, 1);
+        assertEq(tokensList[0], address(dai));
+
+        (
+            ITempleBaseStrategy baseStrategy,
+            uint256 baseStrategyWithdrawalBuffer,
+            uint256 baseStrategyDepositThreshold,
+            ITempleDebtToken dToken
+        ) = trv.borrowTokens(dai);
+        assertEq(address(baseStrategy), address(0));
+        assertEq(baseStrategyWithdrawalBuffer, 100);
+        assertEq(baseStrategyDepositThreshold, 101);
+        assertEq(address(dToken), address(dUSD));
+
+        // Resetting the same token is OK - it just updates the config.
+        {
+            // Base strategy set to something
+            trv.setBorrowToken(dai, bob, 0, 0, address(dUSD));
+            
+            tokensList = trv.borrowTokensList();
+            assertEq(tokensList.length, 1);
+            assertEq(tokensList[0], address(dai));
+
+            (baseStrategy, baseStrategyWithdrawalBuffer, baseStrategyDepositThreshold, dToken) = trv.borrowTokens(dai);
+            assertEq(address(baseStrategy), bob);
+            assertEq(baseStrategyWithdrawalBuffer, 0);
+            assertEq(baseStrategyDepositThreshold, 0);
+            assertEq(address(dToken), address(dUSD));
+        }
+    }
+
+    function test_removeBorrowToken() public {
+        vm.startPrank(executor);
+        vm.expectRevert(abi.encodeWithSelector(ITreasuryReservesVault.BorrowTokenNotEnabled.selector));
+        trv.removeBorrowToken(dai);
+
+        trv.setBorrowToken(dai, address(0), 100, 101, address(dUSD));
+
```

Comment:
```solidity
address[] memory tokensList1 = trv.borrowTokensList();
assertEq(tokensList1.length, 1);
```

Reply:
Fixed in https://github.com/TempleDAO/temple/pull/814/commits/6745674e1060077e4e8093751dd1b3149267d47f